currencies/src/lib.rs

```diff
@@ -218,4 +223,51 @@ where
 
 pub type NativeCurrencyOf<T> = Currency<T, <T as Trait>::GetNativeCurrencyId>;
 
-impl<T: Trait> Module<T> {}
+/// Adapt other currency traits implementation to `BasicCurrency`.
+pub struct BasicCurrencyAdapter<T>(marker::PhantomData<T>);
+
+// Adapat `srml_support::traits::Currency`
+impl<AccountId, T> BasicCurrency<AccountId> for BasicCurrencyAdapter<T>
+where
+	T: SrmlCurrency<AccountId>,
+{
+	type Balance = <T as SrmlCurrency<AccountId>>::Balance;
+	type Error = &'static str;
+
+	fn total_issuance() -> Self::Balance {
+		T::total_issuance()
+	}
+
+	fn balance(who: &AccountId) -> Self::Balance {
+		T::total_balance(who)
+	}
+
+	fn transfer(from: &AccountId, to: &AccountId, amount: Self::Balance) -> result::Result<(), Self::Error> {
+		T::transfer(from, to, amount, ExistenceRequirement::AllowDeath)
+	}
+
+	fn deposit(who: &AccountId, amount: Self::Balance) -> result::Result<(), Self::Error> {
+		let imbalance = T::deposit_creating(who, amount);
+		drop(imbalance);
+
+		Ok(())
+	}
+
+	fn withdraw(who: &AccountId, amount: Self::Balance) -> result::Result<(), Self::Error> {
+		T::withdraw(
+			who,
+			amount,
+			WithdrawReason::Transfer.into(),
+			ExistenceRequirement::AllowDeath,
+		)
+		.map(|imbalance| {
```

Comment:
I don't think you have to call drop, so just `.map(|_| ())` should be good enough